README.md

```diff
@@ -19,8 +19,240 @@
+  Backwards compatible with the old `rd.cos.overlay` stanza.
+
+* `rd.immucore.overlay=LABEL=<vol_label>`: Optionally and mostly for debugging
+  purposes the overlayfs can be mounted on top of a persistent block device.
+  Block devices can be expressed by LABEL (`LABEL=<blk_label>`) or by UUID
+  (`UUID=<blk_uuid>`)
+  Backwards compatible with the old `rd.cos.overlay` stanza.
+
+* `rd.immucore.mount=LABEL:<blk_label>:<mountpoint>`: This option defines a
+  persistent block device and its mountpoint. Block devices can also be
+  defined by UUID (`UUID=<blk_uuid>:<mountpoint>`). This option can be passed
+  multiple times.
+  Backwards compatible with the old `rd.cos.mount` stanza.
+
+* `rd.immucore.oemlabel=<label>`: This option sets the label to search for in order
+  to mount the OEM partition. Defaults to COS_OEM
+  Backwards compatible with the old `rd.cos.oemlabel` stanza.
+
+* `rd.immucore.oemtimeout=<seconds>`: By default we assume the existence of a
+  persistent block device labelled `COS_OEM` which is used to keep some
+  configuration data (mostly cloud-init files). The immutable rootfs tries
+  to mount this device at very early stages of the boot even before applying
+  the immutable rootfs configs. It's done this way to enable to configure the
+  immutable rootfs module within the cloud-init files. As the `COS_OEM` device
+  might not be always present the boot process just continues without failing
+  after a certain timeout. This option configures such a timeout. Defaults to
+  5s.
+  Backwards compatible with the old `rd.cos.oemtimeout` stanza.
+
+* `rd.cos.debugrw`/`rd.immucore.debugrw`: This is a boolean option, true if present, false if not.
+  This option sets the root image to be mounted as a writable device. Note this
+  completely breaks the concept of an immutable root. This is helpful for
+  debugging or testing purposes, so changes persist across reboots.
+
+* `rd.cos.disable`/`rd.immucore.disable`: This is a boolean option, true if present, false if not.
+  It disables the execution of any immutable rootfs module logic at boot.
+
+* `rd.immucore.debug`: Enables debug logging
+
+* `rd.immucore.uki`: Enables UKI booting (Experimental)
+
+
+### Configuration with an environment file
+
+---
+
+The immutable rootfs can be configured with the `/run/cos/cos-layout.env`
+environment file. It is important to note that all the immutable root
+configuration is applied in initrd before switching root and after
+`rootfs` cloud-init stage but before `initramfs` stage. So immutable rootfs
+configuration via cloud-init using the `/run/cos/cos-layout.env` file is
+only effective if called in any of the `rootfs.before`, `rootfs` or
+`rootfs.after` cloud-init stages.
+
+
+In the environment file few options are available:
+
+
+* `VOLUMES=LABEL=<blk_label>:<mountpoint>`: This variable expects a block device
+  and it mountpoint pair space separated list. The default cOS configuration is:
```

Comment:
not totally sure about this one, maybe `a mountpoint`, feel free to adapt or ignore

```suggestion
  and its mountpoint pair space separated list. The default cOS configuration is:
```

Reply:
moutpoint -> mount point.